doc/fr/developper/writestreamconnector.rst

```diff
@@ -0,0 +1,610 @@
+        broker_log:error(1, "event_queue:flush: HTTP POST request FAILED: message line " .. i ..  " is \"" .. v .. "\"")
+      end
+    end
+  
+    -- now that the data has been sent, we empty the events array
+    self.events = {}
+    -- and update the timestamp
+    self.__internal_ts_last_flush = os.time()
+  end
+
+The init() function to get parameters and create the queue
+**********************************************************
+
+In this case, the “init()” function will create the queue with parameters 
+defined by users of use default parameters if some are missing:
+
+.. code-block:: lua
+
+  function init(conf)
+    broker_log:set_parameters(1, "/var/log/centreon-broker/stream-connector-influxdb.log")
+    broker_log:info(2, "init: Beginning init() function")
+    queue = event_queue:new(nil, conf)
+    broker_log:info(2, "init: Ending init() function, Event queue created")
+  end
+
+The write() function to insert event in queue
+*********************************************
+
+The “write()” function is only used to insert filtered events into the queue:
+
+.. code-block:: lua
+
+  function write(e)
+    broker_log:info(3, "write: Beginning write() function")
+    queue:add(e)
+    broker_log:info(3, "write: Ending write() function\n")
+    return true
+  end
+
+The filter() function to select only performance data events
+************************************************************
+To select only performance data, we need to select category 3 (“Storage”) 
+and the element type 1 (“metric”):
+
+.. code-block:: lua
+
+  function filter(category, element)
+    if category == 3 and element == 1 then 
+      return true
+    end
+    return false
+  end
+
+Complete script
+***************
+
+The complete script can by download `here <TODO>`_.
+
+Configure Centreon Broker
+*************************
+
+Configure the new output into Centreon Web interface in 
+**Configuration > Pollers > Broker configuration > Central Broker**. 
+In **Output** tab select **Generic – Stream connector** and click **Add**:
+
+.. image:: /_static/images/developper/lua/add_stream_connector.png
+   :align: center
+
+Define the name of this output and the path to the LUA connector:
+
+.. image:: /_static/images/developper/lua/broker_influxdb_output.png
+   :align: center
+   :scale: 65%
+
+Then click **Save** and go to generate the configuration and restart **cbd**.
+
+.. note::
+   Don’t forget to restart “centengine” too to create the Centreon Broker cache.
+
+If you install the `Grafana <https://grafana.com/>`_ dashboard, you can visualize the stored data:
+
+.. image:: /_static/images/developper/lua/visualize_data_grafana.png
+   :align: center
+   :scale: 65%
+
+Discover other Centreon Stream Connectors
+*****************************************
+
+Centreon provides a Github repository to host LUA scripts developed by Centreon 
+and the community. Please go to the `dedicated Github <http://github.com/centreon/centreon-lua-scripts>`_.
```

Comment:
github repository is : https://github.com/centreon/centreon-stream-connector-scripts